### PR TITLE
V8 style support

### PR DESCRIPTION
**Pull request by ryanhamley**
This PR fixes https://github.com/mapbox/mapbox-gl-language/issues/31 but in doing so, it drops support for all Mapbox styles based on vector tile source v7 as these styles are deprecated and Mapbox is encouraging all users to migrate to v8 styles. This change will require a major version release of this plugin. 

Additionally, this PR drops support for IE11 since Mapbox GL JS is also deprecating support for the browser.

This PR also updates the example pages to use the newest styles and current versions of Mapbox GL JS and the RTL plugin while adding new example pages for Italian, Simplified Chinese and Traditional Chinese.

The code now assumes that your `text-field` will include an expression of the form `['get', 'name_xx']`. This can be a top-level expression, in a `coalesce` expression or nested deeper in an expression. The plugin will also now support Italian (`it`), Simplified Chinese (`zh-Hans`) and Traditional Chinese (`zh-Hant`) by default. v7 style Chinese `zh` is no longer supported.

@mapbox/gl-js

### CODE REVIEW

**Review by samanpwbb — COMMENTED** [review]
@ryanhamley this looks solid! I think we should get the docs updated to clarify requirements for using this library: 
- Streets v8
- Expression text values (no support for curly backets)
- Expects translated languages with coalesce expressions that fall back to non-null `name` field.

I see a few opportunities for improvement but I don't think these are necessary for merge:

1. There is a somewhat common edge case where user will have a text field that isn't already wrapped in a coalesce. Based on the logic implemented now, this could result in a lot of empty labels. To cover these cases, it'd be great if the plugin checked if a field values was wrapped in a coalesce, and added the coalesce if not.
2. To improve developer UX, add some validation to warn on curly bracket tokens, or warn on non-coalesce wrapped get expressions.
3. There are rare edge cases where language get expressions are used in _inputs_ to other expressions rather than outputs. Since the current implementation doesn't know about context when it applies transforms, there's no way to handle these cases. I would at least add a note about this in the readme. Take for example:

```js
["match", 
  ["get", "name"], 
  "Canada", 
  "America's Hat", 
  ["coalesce", 
    ["get", "name_en"], 
    ["get", "name"]
  ]
]
```

File: index.js [hunk 1]
```diff
@@ -152,9 +153,12 @@ function changeLayerTextProperty(isLangField, layer, languageFieldName, excluded
 
 function findStreetsSource(style) {
   var sources = Object.keys(style.sources).filter(function (sourceName) {
-    var source = style.sources[sourceName];
-    return /mapbox-streets-v\d/.test(source.url);
+    var url = style.sources[sourceName].url;
+    // the source URL can reference the source version or the style version
+    // this check and the error forces users to migrate to styles using source version 8
+    return url.indexOf('mapbox.mapbox-streets-v8') > -1 || /mapbox-streets-v[1-9][1-9]/.test(url);
```
Comment (andrewharvey) [inline index.js:159]: geojson sources don't have a url, so should we skip sources here without a url?

Comment (andrewharvey) [inline index.js:159]:
```suggestion
    return url && url.indexOf('mapbox.mapbox-streets-v8') > -1 || /mapbox-streets-v[1-9][1-9]/.test(url);
```

**Review by andrewharvey — APPROVED** [review]
Looks great. I've been running this in production with no issues noticed so far.

**Comment by ryanhamley** [comment]
@samanpwbb I think I addressed all your concerns. This now handles the unwrapped case `text-field: ['get', 'name']` and provides a warning when token syntax is encountered and the documentation has been updated.

@andrewharvey That's great to hear that this is already working well! Once this is merged, I'm going to make a few updates and make a new release.

File: README.md [hunk 2]
```diff
@@ -70,7 +70,18 @@ Create a new [Mapbox GL JS plugin](https://www.mapbox.com/blog/build-mapbox-gl-j
 modifies the layers of the map style to use the `text-field` that matches the browser language.
 
 As of Mapbox GL Language v1.0.0, this plugin no longer supports token values (e.g. `{name}`). v1.0+ expects the `text-field`
-property of a style to use an expression of the form `['get', 'name_en']` or `['get', 'name']`; these expressions can be nested.
+property of a style to use an [expression](https://docs.mapbox.com/mapbox-gl-js/style-spec/expressions/) of the form `['get', 'name_en']` or `['get', 'name']`; these expressions can be nested. Note that `get` expressions used as inputs to other expressions may not be handled by this plugin. For example:
+```
+["match", 
+  ["get", "name"], 
+  "Canada", 
+  "America's Hat",
```
Comment (samanpwbb) [inline README.md:78]: hehe maybe we should use an example that is less offensive to canadians. @tristen do you have suggestions.

**Review by samanpwbb — APPROVED** [review]
Looks good!

File: README.md (hunk 2 above)
Comment (tristen) [reply README.md:78]: I likey. @willymaps?

Comment (ryanhamley) [reply README.md]:
```
["match", 
  ["get", "name"], 
  "California", 
  "Golden State", 
  ["coalesce", 
    ["get", "name_en"], 
    ["get", "name"]
  ]
]
```

**Pull request merged**